src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -109,142 +172,113 @@ public void shouldNotCreateLocationIfConfidential() {
         Location location = generateNewLocation();
         location.setIsConfidential(true);
 
-        int dbSize = locationService.findAll().size();
-        importService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
+        importManager.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, locationService.findAll().size());
+        assertEquals(0, locationService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotUpdateLocationIfConfidential() {
-        Location newLocation = generateNewLocation();
-        newLocation.setIsConfidential(true);
-        generateExistingLocation();
+        Location location = generateExistingLocation();
+        importManager.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
+        assertEquals(1, locationService.findAll().size());
 
-        int dbSize = locationService.findAll().size();
-        Location updated = importService.createOrUpdateLocation(newLocation, EXISTING_EXTERNAL_ID, PROVIDER);
+        location.setIsConfidential(true);
+        location.setName(NEW_STRING);
+        var updated = importManager.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, locationService.findAll().size());
+        assertEquals(1, locationService.findAll().size());
+        assertEquals(EXISTING_STRING, locationService.findAll().get(0).getName());
```

Comment:
So when a location is updated with confidential the old one is still available for other providers and the new one for the owner? In my opinion, this is misleading because the location is outdated.

Reply:
The same physical location for two different providers will be available as two separate entities in the DB. Currently, we're only removing some detached (historical) entities, but we should consider the behavior more accurately for all of them in the future. For now, only entities with OneToMany relations to them are removed when are no longer connected with no other. Those are Languages, Programs, OpeningHours, Contacts, and Phones.